kp.go

```diff
@@ -152,9 +163,19 @@ func NewWithLogger(config ClientConfig, transport http.RoundTripper, logger Logg
 		Logger:      logger,
 		tokenSource: ts,
 	}
+
+	for _, opt := range opts {
+		opt(c)
+	}
 	return c, nil
 }
 
+func WithQSC(qscConfig QSCConfigInfo) Option {
```

Comment:
just leaving comment for context sake. After thinking about this for a while, it does seem we *could* just let the user, post-creation of the client, manually set the QSCConfig in the Client (because it is a exposed struct member), however, in the future, they may be initialization steps to be done at time of construction that we may include in this function. For that reason, I would say leave it this way and we should maybe encourage users to *only* initialize a QSC Client using the option.

Good Example:
```go
client := New(config, transport, WIthQSC(qscConfig))
```

Bad Example:
```go
client :=  New(config, transport)
client.QSCConfig = qscConfig
```

Reply:
Either that or make `QSCClient` a private struct member

Reply:
done